app/src/main/java/com/cdhiraj40/leetdroid/ui/fragments/user/ContestDetailsFragment.kt

```diff
@@ -97,11 +97,11 @@ class ContestDetailsFragment : Fragment(), ContestHistoryAdapter.OnItemClicked {
                             contestHistoryAdapter
 
                         contestHistoryAdapter.setOnClick(this@ContestDetailsFragment)
-                        fragmentContestDetailsBinding.contestHistoryProgressBar.visibility =
-                            View.GONE
+                        loadingView.visibility = View.GONE;
+
                     }
                 } catch (exception: Exception) {
-                    generalErrorView.visibility = View.VISIBLE
+                    loadingView.visibility = View.VISIBLE
```

Comment:
The loading view should not be visible when an error occurs, it should be gone and the error view should be shown which was already there.

```
                    generalErrorView.visibility = View.VISIBLE
                    loadingView.visibility = View.GONE